### DISCUSSION TITLE
ST7735 greenTab settings

### DISCUSSION THREAD

**Discussion by MHz000** [Bodmer/TFT_eSPI#1184] [General]
What is wrong with my setup for a ST7735 greenTab? 
tft.fillScreen(TFT_BLACK) fills just 1/3 of the screen while
tft.print("0.1 0.2 0.5 1K  2K  4K  8K") works fine and writes a whole line.

![IMG_1675](https://user-images.githubusercontent.com/7973146/117563740-3085c700-b0a8-11eb-8567-616ab5809458.JPG)

Thank.you

**Comment by Bodmer** [comment]
It looks like the wrong driver is being used. Run the Read_User_Setup diagnostic sketch and post the results here.

**Comment by MHz000** [comment]
Bodmer first of all I like to thank-you!

My goal is to display an audio spectrum using Arduinos FFT lib on a
somewhat larger color display like the ST7735 green tab.
The environment I'm using is WIN10,  Arduino IDE 1.8.13 and PIO using
ESP8266 or ESP32
However I've to admit my very limited C++ knowledge and almost no PIO
experience :-(

What I've done so far:
1 copy source code e.g. game of live into a new PIO folder/project
2 include bodmer/TFT_eSPI^2.3.67 in platformio.ini
3 include file User_Setup_Select as source code into project folder
4 un-comment line 25 now reading: #include <User_Setups/Setup2_ST7735.h>
//
   (line 17 ... define settings in platformio .. I left active. I don't
understand the hint in "Tools" but I would     like to do as  recommended
there)
5 include file User_Setup.h as source code into project folder
6 edit User_Setup to:
   line 41 #define ST7735_Driver
   line 84 & 85  #define TFT_HEIGHT 320 // ST7789 240 x 320
                        #define TFT_HEIGHT 240 // GC9A01 240 x 240
   line 95 #define ST7735_GREENTAB
7 compile and upload with no problems👍
7 pin definition and wiringBodm seems to be ok as the display show
generation after generation
8 however  inbetween tft.fillScreen(TFT_BLACK); erases just 1/3 of the
screen🤔

Sorry for asking stupid questions
All the best Joschen
------------------------------------------------------------------------------------------------------------------
serial monitor output of running Read_User_Setup as Arduino sketch

[code]
TFT_eSPI ver = 2.2.19
Processor    = ESP8266
Frequency    = 80MHz
Voltage      = 3.32V
Transactions = No
Interface    = SPI
SPI overlap  = No

Display driver = 9341
Display width  = 240
Display height = 320

TFT_CS   = PIN_D-1
TFT_DC   = PIN_D5
TFT_RST  = PIN_D-1
TFT_WR   = PIN_D6
TFT_RD   = PIN_D7

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Font 7      loaded
Font 8      loaded
Smooth font enabled

Display SPI frequency = 20.00
[/code]



Am Do., 13. Mai 2021 um 00:38 Uhr schrieb Bodmer ***@***.***>:

**Reply by TheBlackSheep** [reply]
Are you sure it's this size (320 x 240) - all the ST7735's I've seen are either 128x128 or 128x160 - interestingly one of the versions I have, have a "greentab" on the screen cover but I was seeing an outline (a set of pixels tha weren't being written to within the library) changing to a REDTAB fixed that - my boards (AZ Delivery) have a red pcb like your's and are definitely only 128x160
